src/vendor/vendor.c

```diff
@@ -1834,39 +1809,46 @@ int DiscoverDM_ForAllComponents(char *objs_filename, char *params_filename)
 ** \return  USP_ERR_OK if successful
 **
 **************************************************************************/
-int DiscoverDM_ForComponent(rdk_component_t *rdkc, kv_vector_t *rdk_objects, kv_vector_t *rdk_params)
+int DiscoverDM_ForComponent(kv_vector_t *rdk_objects, kv_vector_t *rdk_params)
 {
-    int i;
-    int ccsp_err;
+    int rbus_err;
     int err;
-    parameterInfoStruct_t **param_infos = NULL;
-    parameterInfoStruct_t *param_info;
-    int num_param_infos = 0;
     char *write_status;
+    rbusElementInfo_t* elem;
+    const char *component, *c1;
 
     // Exit if unable to determine all instances provided by this rdk component
-    USP_LOG_Info("%s: Getting parameters and objects for '%s'", __FUNCTION__, rdkc->group_name);
-    ccsp_err = CcspBaseIf_getParameterNames(bus_handle, rdkc->component_name, rdkc->dbus_path, "Device.", 0, 
-                                          &num_param_infos, &param_infos);
-    if (ccsp_err != CCSP_SUCCESS)
+    USP_LOG_Info("%s: Getting parameters and objects for CCSP components using rbus", __FUNCTION__);
+    rbusElementInfo_t* elems = NULL;
+    rbus_err = rbusElementInfo_get(bus_handle,"Device.", RBUS_MAX_NAME_DEPTH, &elems);
+    if (rbus_err != RBUS_ERROR_SUCCESS)
     {
-        USP_LOG_Error("%s: CcspBaseIf_getParameterNames(%s) failed (%d- %s)", __FUNCTION__, rdkc->group_name, ccsp_err, ToCcspErrString(ccsp_err));
-        return USP_ERR_INTERNAL_ERROR;
+       USP_LOG_Error("%s: rbusElementInfo_get(%s) failed (%d- %s)", __FUNCTION__, elems->name, rbus_err, ToRbusErrString(rbus_err));
+       return USP_ERR_INTERNAL_ERROR;
     }
 
     // Iterate over all parameters and objects found
-    for (i=0; i<num_param_infos; i++)
-    {
-        param_info = param_infos[i];
-        write_status = (param_info->writable) ? "RW" : "RO";
-        Add_NameToDM(rdkc, param_info->parameterName, write_status, rdk_objects, rdk_params);
+    elem = elems;
+    component = NULL;
+    char* component_name = NULL;
+    while(elem)
+    {
+        component = elem->component;
+	c1 = NULL;
+        write_status = (elem->type == RBUS_ELEMENT_TYPE_TABLE ? elem->access & RBUS_ACCESS_ADDROW : elem->access & RBUS_ACCESS_SET)  ? "RW" : "RO";
+        if ((component_name = strrchr(component, '.')))
+              c1 =  component_name+1;
```

Comment:
Missing braces around `c1 = component_name+1;`
But anyway you don't need the name of the component as discussed in an earlier comment, because RBus API doesn't need this info when you call it for get/set/add/delete etc. So it should be removed from the code,

Reply:
component name is required to check,  to overcome below issue 
`DATABASE_Init: Opening database /nvram/usp-pa.db DM_PRIV_AddSchemaPath: Path Device.SoftwareModules.ExecEnv.{i}.CurrentRunLevel already exists in schema USP Agent aborted unexpectedly`
-> usp_dm_objs.conf and usp_dm_params.conf stores Device.SoftwareModules. due to Rbus APIs which should not present .
-> Removed the group argument for Add_ObjectToDM and Add_ParamToDM 
Please review